src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -264,9 +259,9 @@ public override Task<bool> IsSignedAsync(CancellationToken token)
             var isSigned = false;
 
 #if IS_DESKTOP
-            using (var reader = new BinaryReader(_zipStream, _utf8Encoding, leaveOpen: true))
+            if (Zip.GetEntry(_signingSpecifications.SignaturePath) != null)
```

Comment:
This should
1. Get the entry
1. Verify the case of the entry path returned with Ordinal compare

There use to be a helper for this, not sure if we still have it